spotbugs/etc/messages.xml

```diff
@@ -8991,7 +8999,144 @@ Using floating-point variables should not be used as loop counters, as they are
+        it may cause confusion with the java.util.List class from the Java Standard Library.
+        </p>]]>
+    </Details>
+  </BugPattern>
+    <BugPattern type="PI_DO_NOT_REUSE_PUBLIC_IDENTIFIERS_CLASS_NAMES">
+        <ShortDescription>Do not reuse public identifiers from JSL as class name</ShortDescription>
+        <LongDescription>Class name {1} in source file {2} shadows the publicly available identifier from the Java Standard Library.</LongDescription>
+        <Details>
+            <![CDATA[<p>
+      It's essential to avoid reusing public identifiers from the Java Standard Library as class names.
+      This is because the Java Standard Library is a part of the Java platform and is expected to be available in all Java environments.
+      Doing so can lead to naming conflicts and confusion, making it harder to understand and maintain the code.
+      It's best practice to choose unique and descriptive class names that accurately represent the purpose and functionality of your own code.
+      To provide an example, let's say you want to create a class for handling dates in your application. Instead of using a common name like "Date",
+      which conflicts with the existing java.util.Date class, you could choose a more specific and unique name like or "AppDate" or "DisplayDate".
+
+      A few key points to keep in mind when choosing names as identifier:
+      <ul>
+        <li>Use meaningful prefixes or namespaces: Prepend a project-specific prefix or namespace to your class names to make them distinct. For example, if your project is named "MyApp", you could use "MyAppDate" as your class name.</li>
+        <li>Use descriptive names: Opt for descriptive class names that clearly indicate their purpose and functionality. This helps avoid shadowing existing Java Standard Library identifiers. For instance, instead of "List", consider using "CustomAppList".</li>
+        <li>Follow naming conventions: Adhere to Java's naming conventions, such as using camel case (e.g., MyClass) for class names. This promotes code readability and reduces the chances of conflicts.</li>
+      </ul>
+
+      </p>
+      <p>See SEI CERT rule <a href="https://wiki.sei.cmu.edu/confluence/display/java/DCL01-J.+Do+not+reuse+public+identifiers+from+the+Java+Standard+Library">DCL01-J. Do not reuse public identifiers from the Java Standard Library</a>.</p>
+      ]]>
+        </Details>
+    </BugPattern>
+    <BugPattern type="PI_DO_NOT_REUSE_PUBLIC_IDENTIFIERS_FIELD_NAMES">
+        <ShortDescription>Do not reuse public identifiers from JSL as field name</ShortDescription>
+        <LongDescription>The used identifier {2} as field name in class {1} in source file {3} shadows the publicly available identifier from the Java Standard Library.</LongDescription>
+        <Details>
+            <![CDATA[<p>
+      It is crucial to avoid reusing public identifiers from the Java Standard Library as field names in your code.
```

Comment:
"Crucial" is probably putting it too strongly. It's poor practice, but it can work if you're careful, especially if your application doesn't make use of the relevant part of the standard library.

Reply:
@ThrawnCA  I have reconsidered the phrasing and I agree with your point. Thanks for mentioning it!